File: ts/example/src/tests/command_tests.ts
// NOTE: This example does not use any Test Frameworks because Vanilla-Mock can work on any Test Frameworks.

import assert from "assert";
import {ExampleCommandNull, ExampleCommandSpy} from "./command_doubles";
import {ActualCommandUser} from "../command_user";


// EXAMPLE: If the test case do not depend on any command behaviors, you can use **stubs**.
export function testUsingCommandStub() {
  const command = new ExampleCommandNull();

  const sut = new ActualCommandUser(command);

  assert.doesNotThrow(() => {
    sut.doEverything()
  });
}


// EXAMPLE: If the test case should verify indirect output, you can use **spy**.
export function testUsingCommandSpy() {
  const command = new ExampleCommandSpy();

  const sut = new ActualCommandUser(command);

  sut.doEverything();

  assert.strictEqual(command.calledArgs.length, 1);
}

File: swift/example/Tests/VanillaMockExampleTests/CommandTests.swift
import XCTest
@testable import VanillaMockExample



class CommandTests: XCTestCase {
    // EXAMPLE: If the test case do not depend on any command behaviors, you can use **stubs**.
    func testUsingCommandStub() {
        let command = ExampleCommandNull()

        let sut = ActualCommandUser(command: command)

        sut.doEveryThing()
    }


    // EXAMPLE: If the test case should verify indirect output, you can use **spy**.
    func testUsingCommandSpy() {
        let commandSpy = ExampleCommandSpy()

        let sut = ActualCommandUser(command: commandSpy)

        sut.doEveryThing()

        XCTAssertEqual(commandSpy.callArgs.count, 1)
    }
}